tad/api/routes/tasks.py

```diff
@@ -0,0 +1,48 @@
+from typing import Annotated, Any
+
+from fastapi import APIRouter, Depends, Request, status
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+
+from tad.models.task import MoveTask
+from tad.services.tasks import TasksService
+
+router = APIRouter()
+templates = Jinja2Templates(directory="tad/site/templates")
+
+
+@router.post("/move", response_class=HTMLResponse)
+async def move_task(
+    request: Request, move_task: MoveTask, tasks_service: Annotated[TasksService, Depends(TasksService)]
+) -> HTMLResponse:
+    """
+    Move a task through an API call.
+    :param tasks_service: the task service
+    :param request: the request object
+    :param move_task: the move task object
+    :return: a HTMLResponse object, in this case the html code of the card that was moved
+    """
+    try:
+        task = tasks_service.move_task(
+            convert_to_int_if_is_int(move_task.id),
```

Comment:
Pydantic model should do the conversion automaticly if you submit a string and expect an int.

Reply:
I tried several options but could not get it to work how I wanted. The json we receive contains strings, but we need ints in the backend. This is an ugly work-around to get it to work for now. Happy to make it better next round.